From 3771f1d5bd9f059c70d1e56503ef4eb029283eee Mon Sep 17 00:00:00 2001
From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Sun, 25 May 2014 11:57:22 +0200
Subject: [PATCH] keymap: Add Lenovo Enhanced USB Keyboard

https://bugs.freedesktop.org/show_bug.cgi?id=77234
(cherry picked from commit d258d4967eb24122c2b1014d4e873f61b633f1d2)
(cherry picked from commit f36ed0204ee1f917f8eadc37bb655324b1e4157a)

Conflicts:
	hwdb/60-keyboard.hwdb
---
 hwdb/60-keyboard.hwdb | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/hwdb/60-keyboard.hwdb b/hwdb/60-keyboard.hwdb
index fdab8273d9..c320c218ce 100644
--- a/hwdb/60-keyboard.hwdb
+++ b/hwdb/60-keyboard.hwdb
@@ -576,6 +576,17 @@ keyboard:dmi:bvn*:bvr*:bd*:svnLENOVO*:pn*IdeaPad*U300s*:pvr*
  KEYBOARD_KEY_f1=f21
  KEYBOARD_KEY_ce=f20
 
+# enhanced USB keyboard
+keyboard:usb:v04B3p301B*
+ KEYBOARD_KEY_90001=prog1 # ThinkVantage
+ KEYBOARD_KEY_90002=screenlock
+ KEYBOARD_KEY_90003=file
+ KEYBOARD_KEY_90004=wordprocessor
+ KEYBOARD_KEY_90005=spreadsheet
+ KEYBOARD_KEY_90006=calc
+ KEYBOARD_KEY_90007=mail
+ KEYBOARD_KEY_90008=www
+
 ###########################################################
 # Logitech
 ###########################################################
